include/boost/type_index/runtime_cast/detail/runtime_cast_impl.hpp

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (c) Chris Glover, 2016.
+//
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_TYPE_INDEX_RUNTIME_CAST_DETAIL_RUNTIME_CAST_IMPL_HPP
+#define BOOST_TYPE_INDEX_RUNTIME_CAST_DETAIL_RUNTIME_CAST_IMPL_HPP
+
+/// \file pointer_class.hpp
+/// \brief Contains the overload of boost::typeindex::runtime_cast for
+/// pointer types.
+///
+/// boost::typeindex::runtime_cast can be used to emulate dynamic_cast
+/// functionality on platorms that don't provide it or should the user
+/// desire opt in functionality instead of enabling it system wide.
+
+#include <boost/type_index.hpp>
```

Comment:
Currently, boost/type_index.hpp is not including runtime_cast.hpp. If a user is using type_index wants the runtime_cast functionality, I expect that user to include runtime_cast.hpp separately. I personally like finer grained headers like this, but I realize it can be a pain and some people don't like it so I'm happy to change it around so that type_index brings in the basic runtime_cast functionality as well. It's really not a lot of code, more on that below.